changelog/5986.improvement.md

```diff
@@ -0,0 +1 @@
+Using the TrainingDataImporter interface to load the data in core test
```

Comment:
I'd also mention the path in the failed stories file which is absolute now!
```suggestion
Using the `TrainingDataImporter` interface to load the data in `rasa test core` .
```

Reply:
Yes I can do that! 👍🏾 To be honest, it is not quite clear for me what to include in the changelog and what not. Based on the very helpful also advice from @alwx  [here](https://github.com/RasaHQ/rasa-x/pull/4239#discussion_r560809197), I included only the change related to the issue itself. 

@wochinge @alwx Do you think having a guideline on how to write a good changelog will be helpful? There are different articles out there explaining that but maybe having one that outlines what Rasa wants to track and easily find ?

Reply:
>  Based on the README, the changelog is for the users

This is absolutely correct 👍🏻 

> the change in the test should not be mentioned right?

You had to change the test because the behavior of the production code changed. Instead of referring to failed test stories via relative path, Rasa Open Source will now refer to them via absolute path. This is a change which will affect the users and hence it should be covered by a changelog. 

> But I thought that in general a changelog is for the developers no?

**Developers** who are **using** our framework. They are using our framework to train and create contextual AI assistants. They are interested what changes in the usage. Imagine you're a **developer** **using** the `numpy` library. Would you care if they add an additional type annotation? I guess, no. Would you care if they rename `numpy.array` to `numpy.as_array`? I hope yes 😆 

> Otherwise how do we keep track of changes like this one ?

If it's impacting the users we should have a changelog. Otherwise we still have Git (blame) + the commit messages and the related PR to track any changes.

Reply:
Just to note that this Issue [#5986](https://github.com/RasaHQ/rasa/issues/5986) has 2 parts/2 related PRs, one is the current one and the other is this [PR](https://github.com/RasaHQ/rasa/pull/7674). 
The PR (_Implementing the use of `TrainingDataImporter` in `rasa test`_) did not have a changelog but I guess it is not necessary since it did not affect/changed anything on the usage. Just mentioning it here.